HW2/P4/driver.py

```diff
@@ -20,9 +20,76 @@ def py_median_3x3(image, iterations=10, num_threads=1):
     tmpA = image.copy()
     tmpB = np.empty_like(tmpA)
 
+
+    #the original method provided by professors
+    # for i in range(iterations):
+    #    filtering.median_3x3(tmpA, tmpB, 0, 1)
+    #    # swap direction of filtering
+    #    tmpA, tmpB = tmpB, tmpA
+    #
+    # #multiple-threads method
+    # #first,create an event for each (threads,iteration)
+    # #event will be used for protecting communications between threads
+    Events = []
+    for x in range(num_threads):
+        for iter in range(iterations):
+            Events.append(threading.Event())
+    #reshape the events, and access it in the future
+    Events = np.array(Events).reshape((num_threads,iterations))
```

Comment:
It's unclear to me immediately what this code does or why, it might be helpful to explain your events structure in the comments. Once I figured out what you were doing though, the rest of the code made a lot more sense.